metagraph/src/graph/representation/masked_graph.hpp

```diff
@@ -113,14 +109,24 @@ class MaskedDeBruijnGraph : public DeBruijnGraph {
     virtual bool operator==(const MaskedDeBruijnGraph &other) const;
     virtual bool operator==(const DeBruijnGraph &other) const override;
 
-    virtual void set_mask(bitmap *mask) { kmers_in_graph_.reset(mask); }
+    virtual void set_mask(std::unique_ptr<bitmap>&& mask) {
+        assert(mask.get());
+        assert(mask->size() == graph_->max_index() + 1);
+        kmers_in_graph_ = std::move(mask);
+    }
+
+    typedef std::function<void(const std::function<void(node_index, bool)>&)> GenerateNodes;
+    virtual void update_mask(const GenerateNodes &generate_nodes,
```

Comment:
not clear what that `bool` represents
```suggestion
    typedef std::function<void(const std::function<void(node_index, bool /* ??? */)>&)> GenerateNodes;
    virtual void update_mask(const GenerateNodes &generate_nodes,
```